edx_dl/edx_dl.py

```diff
@@ -701,29 +736,34 @@ def download_url(url, filename, headers, args):
     if is_youtube_url(url):
         download_youtube_url(url, filename, headers, args)
     else:
-        import ssl
-        # FIXME: Ugly hack for coping with broken SSL sites:
-        # https://www.cs.duke.edu/~angl/papers/imc10-cloudcmp.pdf
-        #
-        # We should really ask the user if they want to stop the downloads
-        # or if they are OK proceeding without verification.
-        #
-        # Note that skipping verification by default could be a problem for
-        # people's lives if they happen to live ditatorial countries.
-        #
-        # Note: The mess with various exceptions being caught (and their
-        # order) is due to different behaviors in different Python versions
-        # (e.g., 2.7 vs. 3.4).
         try:
-            urlretrieve(url, filename)
-        except Exception as e:
-            logging.warn('Got SSL/Connection error: %s', e)
-            if not args.ignore_errors:
-                logging.warn('Hint: if you want to ignore this error, add '
-                             '--ignore-errors option to the command line')
-                raise e
-            else:
-                logging.warn('SSL/Connection error ignored: %s', e)
+            bin = 'wget'
+            cmd = [bin, url, '-c', '-O', unquote(filename), '--no-cookies', '--no-check-certificate']
+            execute_command(cmd, args)
```

Comment:
Please, note that while I would love to see all the code related to downloading removed from the program and handed over to other utilities, we can't be sure that the users environments are such that external programs are available.

Even if the programs are available on their particular platform, they may not have the permission to install such programs (e.g., lack of administrator privileges) or not have them in their PATHs or whatever.